syslog-ng-ctl/syslog-ng-ctl.c

```diff
@@ -419,6 +424,9 @@ slng_passwd(int argc, char *argv[], const gchar *mode, GOptionContext *ctx)
           return 1;
         }
 
+      if (!credentials_secret && credentials_remaining && credentials_remaining[remaining_unused_index])
```

Comment:
the same here: factor out a subexpr, or the whole expr. to a func.

Reply:
Fixed